### PR TITLE
fix(deps): fixing included depenednecies

### PR DESCRIPTION
**Pull request by bassrock**
# Goal

The peer dependencies were in the devDependencies  when they should have been in the dependencies block. This will include them on install for all downstream packages for use.

This also does the following:
* Cleans up GitHub workflows and names
* Cleans up dependabot config file and commit message prefixes
  * Different commit message for when dev dependencies are updated vs. regular dependencies so a release is only triggered for regular dependencies (will trigger patch release)
* Enables dependabot auto-merging
* Updates terraform pre-built packages
* Adds missing exports for files we forgot to export
* Changes repository URL to the HTTP endpoint.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,23 +1,29 @@
 // this is the entry point for the npm package
 // anything we want consumable (module, type, class, etc) should be exported here
 export * from './pocket/PocketALBApplication';
-export * from './pocket/PocketVPC';
+export * from './pocket/PocketECSCodePipeline';
```
Comment (bassrock) [inline src/index.ts:4]: This is just a re-alphabetizing and exporting files that were forgotten

File: .github/workflows/publish_github.yml [hunk 2]
```diff
@@ -1,15 +1,12 @@
-name: Node.js Github Package
+name: Publish
 
 on:
-  pull_request:
-    branches:
-      - main
-  push:
+  pull_request_target:
```
Comment (bassrock) [inline .github/workflows/publish_github.yml:4]: we need to use pull_request_target due to this blog post https://github.blog/changelog/2021-02-19-github-actions-workflows-triggered-by-dependabot-prs-will-run-with-read-only-permissions/

File: .github/workflows/publish_github.yml [hunk 3]
```diff
@@ -1,15 +1,12 @@
-name: Node.js Github Package
+name: Publish
```
Comment (kkyeboah) [inline .github/workflows/publish_github.yml:1]: This is specifically publishing to GitHub registry, let's make it clear in the name.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: hashicorp/setup-terraform@v1.3.2
+        with:
+          terraform_version: 0.13.5
+          terraform_wrapper: false
+      - uses: actions/setup-node@v2.3.0
+        with:
+          node-version: '16.x'
+          registry-url: 'https://npm.pkg.github.com'
```
Comment (kkyeboah) [inline .github/workflows/test.yml:20]: I don't think all the registry options here are required for a test or a build

**Review by kkyeboah — COMMENTED** [review]
Nice catch on the peer dependencies! Also, thanks for cleaning up the workflows. I have a few comments and a small request since you are making updates. Could you include the change in this PR https://github.com/Pocket/terraform-modules/pull/244?

**Review by kkyeboah — DISMISSED** [review]
I made these review changes, someone else should definitely review this. My approval means nothing here!!!

**Review by nina-py — DISMISSED** [review]
We paired on this with Kelvin! :)

File: .github/dependabot.yml [hunk 5]
```diff
@@ -11,30 +11,18 @@ updates:
     schedule:
```
Comment (kschelonka) [inline .github/dependabot.yml:11]: I'm not so familiar with this -- could someone explain the changes on this file for me?

File: .github/workflows/linter.yml [hunk 6]
```diff
@@ -11,13 +11,13 @@ name: Lint Code Base
...
-# Start the job on all push #
-#############################
+####################################
+# Start the job on all prs to main #
+####################################
 on:
-  push:
-    branches-ignore:
-      - 'main'
+  pull_request:
```
Comment (kschelonka) [inline .github/workflows/linter.yml:18]: 👍 main is protected from push, yes?

**Review by kschelonka — DISMISSED** [review]
I'm a little unclear on the dependabot changes (so if anyone is to explain I'd be grateful 🙏 ), but I think given the number of eyes on it this seems good to me :)

File: .github/workflows/linter.yml (hunk 6 above)
Comment (kkyeboah) [reply .github/workflows/linter.yml:18]: yes

File: .github/dependabot.yml [hunk 7]
```diff
@@ -11,30 +11,18 @@ updates:
...
-  - package-ecosystem: 'docker'
-    # Look for a `Dockerfile` in the `root` directory
-    directory: '/'
-    # Check for updates once a week
-    schedule:
-      interval: 'daily'
-    commit-message:
-      prefix: '[DEPBOT-0102]'
+      #Packaged dependencies (release)
+      prefix: fix
```
Comment (kkyeboah) [inline .github/dependabot.yml:15]: update dependabot PR title prefix so when a PR is approved and merged, a new release is created.

File: .github/dependabot.yml [hunk 8]
```diff
@@ -11,30 +11,18 @@ updates:
...
   # Maintain dependencies for GitHub Actions
   - package-ecosystem: 'github-actions'
     directory: '/'
     schedule:
       interval: 'daily'
     commit-message:
-      prefix: '[DEPBOT-0103]'
-
-  # Maintain dependencies for Terraform
-  - package-ecosystem: 'terraform'
```
Comment (kkyeboah) [inline .github/dependabot.yml:35]: remove terraform because there's no HCL here

**Review by kkyeboah — APPROVED** [review]

**Pull request merged**